sklearn/neighbors/base.py

```diff
@@ -347,6 +518,7 @@ class from an array representing our data set and ask who's
             )
         n_samples, _ = X.shape
         sample_range = np.arange(n_samples)[:, None]
+        neigh_dist = None
```

Comment:
If you're going to do this (to make lgtm.com happy?), you should also add `assert neigh_dist is not None` before returning. Before, a logic error would have raised a NameError; now we are unprotected.